pulsar-broker/src/main/java/org/apache/pulsar/broker/s3offload/impl/BlockAwareSegmentInputStream.java

```diff
@@ -90,68 +84,60 @@ private int readEntries() throws IOException {
         checkState(bytesReadOffset >= DataBlockHeaderImpl.getDataStartOffset());
         checkState(bytesReadOffset < dataBlockFullOffset);
 
-        try {
-            // once reach the end of entry buffer, start a new read.
-            if (entriesByteBuf.isEmpty()) {
-                readNextEntriesFromLedger();
-                log.debug("After readNextEntriesFromLedger: bytesReadOffset: {}, blockBytesHave: {}",
-                        bytesReadOffset, blockBytesHave);
-            }
-
-            // always read from the first ByteBuf in the list, once read all of its content remove it.
-            ByteBuf entryByteBuf = entriesByteBuf.get(0);
-            int ret = entryByteBuf.readByte();
-            bytesReadOffset ++;
+        // once reach the end of entry buffer, start a new read.
+        if (entriesByteBuf.isEmpty()) {
+            entriesByteBuf = readNextEntriesFromLedger(startEntryId + blockEntryCount, ENTRIES_PER_READ);
```

Comment:
what happens if readNextEntriesFromLedger cannot read any more entries?

Reply:
it will return empty list, and line 93 will get IndexOutOfBoundsException.